yammbs/analysis.py

```diff
@@ -152,35 +150,126 @@ def get_internal_coordinate_rmsds(
+                key = tuple(
+                    (
+                        internal_coordinate.a,
+                        internal_coordinate.b,
+                        internal_coordinate.c,
+                    ),
+                )
+
+            if isinstance(internal_coordinate, Dihedral):
+                key = tuple(
+                    (
+                        internal_coordinate.a,
+                        internal_coordinate.b,
+                        internal_coordinate.c,
+                        internal_coordinate.d,
+                    ),
+                )
+
+            if isinstance(internal_coordinate, OutOfPlane):
+                # geomeTRIC lists the central atom FIRST, but SMIRNOFF force fields list
+                # the central atom SECOND. Re-ordering here to be consistent with SMIRNOFF
+                # see PR #109 for more
+
+                key = tuple(
+                    (
+                        internal_coordinate.b,  # NOTE!
+                        internal_coordinate.a,  # NOTE!
+                        internal_coordinate.c,
+                        internal_coordinate.d,
+                    ),
+                )
 
-    internal_coordinates = {
-        label: [
-            (
-                internal_coordinate.value(reference),
-                internal_coordinate.value(target),
+            key = tuple(int(index) for index in key)
+
+            internal_coordinates[label].update(
+                {
+                    key: (
+                        internal_coordinate.value(reference),
+                        internal_coordinate.value(target),
+                    ),
+                },
             )
-            for internal_coordinate in _generator.Internals
-            if isinstance(internal_coordinate, internal_coordinate_class)  # type: ignore[arg-type]
-        ]
-        for label, internal_coordinate_class in types.items()
-    }
+
+    return internal_coordinates
+
+
+def get_internal_coordinate_differences(
+    molecule: Molecule,
+    reference: Array,
+    target: Array,
+    _types: tuple[str, ...] = ("Bond", "Angle", "Dihedral", "Improper"),
+) -> dict[str, dict[tuple[int, ...], float]]:
+    differences: dict[str, dict[tuple[int, ...], float]] = dict()
+
+    internal_coordinates = get_internal_coordinates(
+        molecule=molecule,
+        reference=reference,
+        target=target,
+        _types=_types,
+    )
+
+    for label, values_with_indices in internal_coordinates.items():
+        differences[label] = dict()
+
+        for indices, values in values_with_indices.items():
+            differences[label][indices] = values[1] - values[0]
+
+    return differences
+
+
+def get_internal_coordinate_rmsds(
+    molecule: Molecule,
+    reference: Array,
+    target: Array,
+    _types: tuple[str, ...] = ("Bond", "Angle", "Dihedral", "Improper"),
+) -> dict[str, float]:
+    """Get internal coordinate RMSDs for one conformer of one molecule."""
```

Comment:
I think expanded docstrings would be beneficial. The typing shows that `molecule` is class type `Molecule` but I would have to look at the source code to know it was an `openff_toolkit.Molecule`.  Also that `reference` and `target` are both conformers in `molecules` and that all conformers in `molecule` will be discarded, and replaced with either `reference` or `target`.